Client/lib/gMap.js

```diff
@@ -0,0 +1,282 @@
+
+  var gMap = {}
```

Comment:
It would be more idiomatic to use an object literal here and assign null to any values you do not have valued for yet.